cmd/cut-release/main.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/plaid/go-envvar/envvar"
+)
+
+type envVars struct {
+	// Version is the new release version to use
+	Version string `envvar:"VERSION" default:"development"`
+}
+
+func main() {
+	env := envVars{}
+	if err := envvar.Parse(&env); err != nil {
+		log.Fatal(err)
+	}
+
+	updateHardCodedVersions(env.Version)
+
+	generateTypescriptClientDocs()
+}
+
+func generateTypescriptClientDocs() {
+	// Run `yarn install` to make sure `TypeDoc` dep is installed
+	cmd := exec.Command("yarn", "install")
+	cmd.Dir = "rpc/clients/typescript"
+	if err := cmd.Run(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Run `yarn docs:md` to generate MD docs
+	cmd = exec.Command("yarn", "docs:md")
+	cmd.Dir = "rpc/clients/typescript"
+	if err := cmd.Run(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Update SUMMARY.md
+	tsClientSummaryPath := "docs/json_rpc_clients/typescript/SUMMARY.md"
+	dat, err := ioutil.ReadFile(tsClientSummaryPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Modify the paths to be prefixed with `json_rpc_clients/typescript`
+	modifiedDat := strings.Replace(string(dat), "](", "](json_rpc_clients/typescript/", -1)
+	modifiedDat = strings.Replace(modifiedDat, "](json_rpc_clients/typescript/)", "]()", -1)
+	finalTsClientSummary := strings.Replace(modifiedDat, "* [", "  * [", -1)
+
+	// Replace the summary content nested under `Typescript client`
+	mainSummaryPath := "docs/SUMMARY.md"
+	dat, err = ioutil.ReadFile(mainSummaryPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	finalSummaryLines := []string{}
+	lines := strings.Split(string(dat), "\n")
+	isCutting := false
+	for _, l := range lines {
+		if strings.Contains(l, "## Contributing") {
+			isCutting = false
+			finalSummaryLines = append(finalSummaryLines, finalTsClientSummary, "")
+		}
+		if !isCutting {
+			finalSummaryLines = append(finalSummaryLines, l)
+		}
+		if strings.Contains(l, "[Typescript client]") {
+			isCutting = true
+		}
+	}
+	finalSummary := strings.Join(finalSummaryLines, "\n")
+	err = ioutil.WriteFile(mainSummaryPath, []byte(finalSummary), 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Remove the nested SUMMARY.MD file
+	err = os.Remove(tsClientSummaryPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// Update the version string in all files that must be updated for a new release
+func updateHardCodedVersions(version string) {
+
+	// Update `rpc/clients/typescript/package.json`
+	tsClientPackageJSONPath := "rpc/clients/typescript/package.json"
+	updateVersionKey(tsClientPackageJSONPath, version)
+
+	// Update `core.ts`
+	corePath := "core/core.go"
+	updateVersionKey(corePath, version)
+
+	// Update badge in README.md
+	readmePath := "README.md"
+	updateBadge(readmePath, version)
+}
+
+func updateVersionKey(filePath string, version string) {
+	dat, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	newVersionString := fmt.Sprintf(`"version": "%s",`, version)
```

Comment:
If we follow the same flow that we have been using, `"version"` will be the previously released version (e.g. `"1.0.6-beta"`), not `"development"`. Since we don't know what the previous version was, we might need to use a regex to find/replace what we're looking for instead of a hardcoded string.